zcash_client_backend/src/welding_rig.rs

```diff
@@ -88,10 +90,23 @@ fn scan_output<P: consensus::Parameters, K: ScanningKey>(
 /// A key that can be used to perform trial decryption and nullifier
 /// computation for a Sapling [`CompactOutput`]
 ///
+/// The purpose of this trait is to enable [`scan_block`]
+/// and related methods to be used with either incoming viewing keys
+/// or full viewing keys, with the data returned from trial decryption
+/// being dependent upon the type of key used. In the case that an
+/// incoming viewing key is used, only the note and payment address
+/// will be returned; in the case of a full viewing key, the
+/// nullifier for the note can also be obtained.
+///
 /// [`CompactOutput`]: crate::proto::compact_formats::CompactOutput
+/// [`scan_block`]: crate::welding_rig::scan_block
 pub trait ScanningKey {
+    /// The type of nullifier extracted when a note is successfully
+    /// obtained by trial decryption.
     type Nf;
 
+    /// Attempt to decrypt a Sapling note and payment address
```

Comment:
Per the Rust style guide:
```suggestion
    /// Attempts to decrypt a Sapling note and payment address
```